src/Layouts/Partials/TransformPanel.vala

```diff
@@ -0,0 +1,85 @@
+/*
+* Copyright (c) 2018 Alecaddd (http://alecaddd.com)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Ana Gelez <ana@gelez.xyz>
+*/
+public class Akira.Layouts.Partials.TransformPanel : Gtk.Grid {
+  construct {
+    border_width = 12;
+    row_spacing = 6;
+    column_spacing = 6;
+		hexpand = true;
+
+    attach (group_title (_("Position")), 0, 0);
+    attach (new Akira.Widgets.LinkedInput (C_("The horizontal coordinate", "X")), 0, 1, 1);
+    attach (new Akira.Widgets.LinkedInput (C_("The vertical coordinate", "Y")), 2, 1, 2);
+
+    attach (group_title (_("Size")), 0, 2);
+    attach (new Akira.Widgets.LinkedInput (C_("The first letter of Width", "W")), 0, 3, 1);
+    var lock_changes = new Gtk.Button.from_icon_name ("changes-allow-symbolic");
+    lock_changes.get_style_context ().add_class ("flat");
+    attach (lock_changes, 1, 3);
```

Comment:
Would you be able to add a bit of reactivity to the lock?
If clicked it changes from `allow` to `disallow`, and if is disallowed `W` and `H` values should maintain the current ratio.